internal/security/spiffetokenprovider/main.go

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright 2022 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package spiffetokenprovider
+
+import (
+	"context"
+	"os"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/security/spiffetokenprovider/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/spiffetokenprovider/container"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/flags"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/interfaces"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+)
+
+func Main(ctx context.Context, cancel context.CancelFunc) {
+	startupTimer := startup.NewStartUpTimer("security-spiffe-token-provider")
+
+	var insecureSkipVerify bool
+	var vaultInterval int
+
+	// All common command-line flags have been moved to bootstrap. Service specific flags are add here,
+	// but DO NOT call flag.Parse() as it is called by bootstrap.Run() below
+	// Service specific used is passed below.
+	f := flags.NewWithUsage(
+		"    --insecureSkipVerify=true/false Indicates if skipping the server side SSL cert verification, similar to -k of curl\n" +
+			"    --vaultInterval=<seconds>       Indicates how long the program will pause between vault initialization attempts until it succeeds",
+	)
+
+	if len(os.Args) < 2 {
+		f.Help()
+	}
+
+	f.FlagSet.BoolVar(&insecureSkipVerify, "insecureSkipVerify", false, "")
+	f.FlagSet.IntVar(&vaultInterval, "vaultInterval", 30, "")
+	f.Parse(os.Args[1:])
```

Comment:
I stole the wrong main.   These command line args are not appropriate.  Need a Main() that better resembles what a normal EdgeX service uses.